torch/_inductor/codecache.py

```diff
@@ -249,35 +253,151 @@ def code_hash(code):
     )
 
 
-def get_code_path(source_code: Union[str, bytes], ext: str, extra: str, binary=False):
-    if binary:
-        source_code = repr(source_code)
-    basename = code_hash(source_code + extra)
+def serialize_fx_arg(arg: Any):
+    if isinstance(arg, torch.fx.GraphModule):
+        # Special case for fx_graph
+        return arg.print_readable(print_output=False).encode("utf-8")
+    if isinstance(arg, list):
+        if len(arg) > 0 and isinstance(arg[0], torch.Tensor):
+            # Special case for example_inputs, everything in torch/csrc/dynamo/guards.cpp
+            # Except for requires_grad, as that has already been handled
+            pickled_guards = []
+            for t in arg:
+                pickled_guards.append(
+                    b"".join(
+                        [
+                            pickle.dumps(torch._C._dispatch_keys(t).__repr__()),
+                            pickle.dumps(t.dtype),
+                            pickle.dumps(t.device),
+                            pickle.dumps(t.shape),
+                        ]
+                    )
+                )
+            return b"".join(pickled_guards)
+    return pickle.dumps(arg)
+
+
+def compiled_fx_graph_hash(fx_args: List[Any]):
+    serialized_fx_args = b"".join([serialize_fx_arg(arg) for arg in fx_args])
+    hashed_fx_args = base64.b32encode(hashlib.sha256(serialized_fx_args).digest())[:51]
+    return "f" + hashed_fx_args.decode("utf-8").lower()
+
+
+def get_path(basename: str, extension: str):
```

Comment:
super nit: but for path annotations prefer `os.PathLike`

Reply:
do you mean this for the basename parameter? it should just be the name itself with no  paths involved, but I guess it technically can be a PathLike object too